### DISCUSSION TITLE
Unable to install brew (uninitialized constant URL::T (NameError))

### DISCUSSION THREAD

**Discussion by sudhakolavali** [Homebrew/discussions#56] [Everyday usage]
Running any brew command threw this error. So I tried to uninstall and reinstall brew.
Installing brew doesnt succeed.

here is the log:

/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
Password:
==> This script will install:
/usr/local/bin/brew
/usr/local/share/doc/homebrew
/usr/local/share/man/man1/brew.1
/usr/local/share/zsh/site-functions/_brew
/usr/local/etc/bash_completion.d/brew
/usr/local/Homebrew

Press RETURN to continue or any other key to abort
==> Downloading and installing Homebrew...
HEAD is now at b8a93279f Merge pull request #8905 from jonchang/bintray-dont-error-on-matching-files
Traceback (most recent call last):
	21: from /usr/local/Homebrew/Library/Homebrew/brew.rb:30:in `<main>'
	20: from /usr/local/Homebrew/Library/Homebrew/brew.rb:30:in `require_relative'
	19: from /usr/local/Homebrew/Library/Homebrew/global.rb:51:in `<top (required)>'
	18: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	17: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	16: from /usr/local/Homebrew/Library/Homebrew/cli/args.rb:3:in `<top (required)>'
	15: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	14: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	13: from /usr/local/Homebrew/Library/Homebrew/cli/named_args.rb:5:in `<top (required)>'
	12: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	11: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	10: from /usr/local/Homebrew/Library/Homebrew/cask/cask_loader.rb:4:in `<top (required)>'
	 9: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	 8: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	 7: from /usr/local/Homebrew/Library/Homebrew/cask/cask.rb:5:in `<top (required)>'
	 6: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	 5: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	 4: from /usr/local/Homebrew/Library/Homebrew/cask/dsl.rb:23:in `<top (required)>'
	 3: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	 2: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	 1: from /usr/local/Homebrew/Library/Homebrew/cask/url.rb:7:in `<top (required)>'
/usr/local/Homebrew/Library/Homebrew/cask/url.rb:8:in `<class:URL>': uninitialized constant URL::T (NameError)
Failed during: /usr/local/bin/brew update --force

**Comment by SMillerDev** [comment]
See #53

**Reply by sudhakolavali** [reply]
Not sure I understand the response there. How do I get a list of all system gems that I need to delete? Also, I am a newbie to brew.

**Comment by sudhakolavali** [comment]
Figured it out. Here are the steps for newbie's like me:
1. run `gem env`
2. get Gem paths and list gems in the path
3. gem uninstall --all
4. run install brew command

**Comment by Rylan12** [comment]
For anyone else who may come across this issue, I found that a full reinstallation of Homebrew was not necessary.

For me, simply installing and uninstalling the `sorbet-runtime` gem for system ruby was enough to solve the problem (possibly in conjunction with https://github.com/Homebrew/brew/pull/8909)

**Reply by hoolatech** [reply]
Not work for me.
